### DISCUSSION TITLE
Is it possible to have different completion styles for corfu and everything else?

### DISCUSSION THREAD

**Discussion by sbancuz** [minad/corfu#513] [Q&A] [answered]
Hi I set up corfu with orderless like this
```
(use-package corfu
  :custom
  (corfu-cycle t)         
  (corfu-auto        t)
  (corfu-auto-delay  0.3)
  (corfu-auto-prefix 1)
  (corfu-quit-no-match 'separator)
  (corfu-popupinfo-delay 0.3)
  (completion-styles '(orderless-fast basic))
  :hook (prog-mode . corfu-mode)
  :init
  (global-corfu-mode)
  (corfu-popupinfo-mode 1))

(defun orderless-fast-dispatch (word index total)
  (and (= index 0) (= total 1) (length< word 4)
       (cons 'orderless-literal-prefix word)))

(orderless-define-completion-style orderless-fast
  (orderless-style-dispatchers '(orderless-fast-dispatch))
  (orderless-matching-styles '(orderless-literal orderless-regexp)))
```

But I was wondering, is it possible to use the fast dispatcher only for the in buffer text completions (the ones where you are typing in a buffer and you get lsp symbols, I don't know how it's actually called) and for everything else, like consult, use some other better dispatchers?

**Comment by minad** [comment, marked as answer]
No, unfortunately this is not easily possible.

EDIT: To expand - I wanted to have this functionality myself a while ago, but overall Orderless just works well enough in all scenarios such that the need was never strong enough. Note that you could configure a separate completion style for the lsp completion category only.
